src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -992,5 +993,34 @@ public static bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> secon
                 value, comparer);
             return BinarySearch(span, comparable);
         }
+
+        /// <summary>
+        /// Creates an <see cref="IEnumerable{T}"/> view of the given <paramref name="memory" /> to allow
+        /// the <paramref name="memory" /> to be used in existing APIs that take an <see cref="IEnumerable{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The element type of the <paramref name="memory" />.</typeparam>
+        /// <param name="memory">The Memory to view as an <see cref="IEnumerable{T}"/></param>
+        /// <returns>An <see cref="IEnumerable{T}"/> view of the given <paramref name="memory" /></returns>
+        public static IEnumerable<T> ToEnumerable<T>(Memory<T> memory) => new MemoryEnumerable<T>(memory);
+
+        internal class MemoryEnumerable<T> : IEnumerable<T>, IEnumerable, IEnumerator<T>, IEnumerator, IDisposable
+        {
+            ReadOnlyMemory<T> _memory;
+            int _index;
+
+            public MemoryEnumerable(Memory<T> memory)
+            {
+                _memory = memory;
+                _index = -1;
+            }
+
+            T IEnumerator<T>.Current => _memory.Span[_index];
+            object IEnumerator.Current => _memory.Span[_index];
+            void IDisposable.Dispose() { }
+            IEnumerator<T> IEnumerable<T>.GetEnumerator() => this;
```

Comment:
This is broken if GetEnumerator is called twice on the same instance and then both enumerators are used interleaved or concurrently.

Reply:
Oh yeah, sorry, this implementation I have right now is garbage. I was testing a super naive non-iterator implementation but stopped midway to continue the discussion on whether it should be an extension method.

Reply:
Ok :)